src/main/java/io/vertx/ext/jdbc/impl/actions/JDBCSQLRowStream.java

```diff
@@ -169,10 +176,9 @@ private void readRow(Future<JsonArray> fut) {
             result.addNull();
           }
         }
-        fut.complete(result);
-      } else {
-        fut.complete();
+        accumulator.add(result);
       }
+      fut.complete();
```

Comment:
@vietj It seems `paused.compareAndSet(false, false)` is more appropriate here? We should keep existing `paused` value if it was changed while executeBlocking runs